tools/integration_tests/util/mounting/multiple_mounting/multiple_mounting.go

```diff
@@ -0,0 +1,114 @@
+//Copyright 2023 Google Inc. All Rights Reserved.
+//
+//Licensed under the Apache License, Version 2.0 (the "License");
+//you may not use this file except in compliance with the License.
+//You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+//Unless required by applicable law or agreed to in writing, software
+//distributed under the License is distributed on an "AS IS" BASIS,
+//WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//See the License for the specific language governing permissions and
+//limitations under the License.
+
+package multiple_mounting
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/mounting"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/operations"
+	"github.com/googlecloudplatform/gcsfuse/tools/integration_tests/util/setup"
+)
+
+func mountMultipleGCSfuseMounts(flags []string, mountPaths []string) (err error) {
+	commonFlags := []string{"--debug_gcs",
+		"--debug_fs",
+		"--debug_fuse",
+		// With this setup, logs of all mounts will be written to same log file.
+		"--log-file=" + setup.LogFile(),
+		"--log-format=text",
+		setup.TestBucket()}
+
+	for i := 0; i < len(commonFlags); i++ {
+		flags = append(flags, commonFlags[i])
+	}
+
+	// Unmount the mounts in case of failure mounting any one of the mounts.
+	shouldUnMount := false
+	unMount := func(mountPath string) {
+		if shouldUnMount {
+			err := mounting.UnMountGcsfuse(mountPath)
+			if err != nil {
+				log.Printf("failure in unmounting: %s\n", err.Error())
+			}
+			operations.RemoveDir(mountPath)
+			return
+		}
+	}
+
+	for i := 0; i < len(mountPaths); i++ {
+		// Add mount path to flags
+		mountFlags := make([]string, len(flags)+1)
+		copy(mountFlags, flags)
+		mountFlags[len(flags)] = mountPaths[i]
+		err := os.Mkdir(mountPaths[i], 0755)
+		if err != nil {
+			shouldUnMount = true
+			return err
+		}
+		err = mounting.MountGcsfuse(setup.BinFile(), mountFlags, setup.LogFile())
+		if err != nil {
+			shouldUnMount = true
+			return err
+		}
+		defer unMount(mountPaths[i])
+	}
+
+	return err
+}
+
+func executeTestsForMultipleMounting(flags [][]string, mountPaths []string, m *testing.M) (successCode int) {
+	var err error
+
+	for i := 0; i < len(flags); i++ {
+		setup.CleanMntDir()
+		if err = mountMultipleGCSfuseMounts(flags[i], mountPaths); err != nil {
+			setup.LogAndExit(fmt.Sprintf("mountMultipleGCSfuseMounts: %v\n", err))
+		}
+
+		successCode = setup.ExecuteTest(m)
+
+		var unmountingError error
+		for i := 0; i < len(mountPaths); i++ {
+			err := mounting.UnMountGcsfuse(mountPaths[i])
+			if err != nil {
+				unmountingError = err
```

Comment:
incase multiple unmounts failed, can we print the error here and do exit later?

Reply:
done